mlrun/api/db/sqldb/db.py

```diff
@@ -665,6 +674,204 @@ def delete_project(self, session, name: str):
+                f.type = "feature"
+                fs.features.append(f)
+
+    def add_feature_set(self, session, project, fs: dict, versioned=False):
+        fs = fs.copy()
+
+        fs_meta = fs["metadata"]
+        name = fs_meta["name"]
+        if not name or not project:
+            raise ValueError("feature-set missing name or project")
+
+        self._ensure_project(session, project)
+        tag = fs_meta.pop("tag", None) or "latest"
+
+        hash_key = fill_function_hash(fs, tag)
+
+        # ??? - is this needed???
+        # # clear tag from object in case another function will "take" that tag
+        # update_in(function, "metadata.tag", "")
+
+        if versioned:
+            uid = hash_key
+        else:
+            uid = f"unversioned-{tag}"
+
+        feature_set = self._query(
+            session, FeatureSet, name=name, project=project, uid=uid
+        ).one_or_none()
+        if not feature_set:
+            state = get_in(fs, "status.state")
+
+            feature_set = FeatureSet(
+                name=name,
+                project=project,
+                state=state,
+                status=get_in(fs, "status"),
+                uid=uid,
+                updated=datetime.now(timezone.utc),
+            )
+
+        fs_spec = fs.get("spec")
+        if fs_spec:
+            features = fs_spec.pop("features", [])
+            entities = fs_spec.pop("entities", [])
+
+        self._update_fs_features(feature_set, features, is_entity=False)
+        self._update_fs_features(feature_set, entities, is_entity=True)
+
+        labels = fs_meta.pop("labels", {})
+        update_labels(feature_set, labels)
+
+        self._upsert(session, feature_set)
+        self.tag_objects_v2(session, [feature_set], project, tag)
+
+        return hash_key
+
+    def update_feature_set(self, session, project, name, fs: dict, tag=None, uid=None):
+        # If object uid was given, use it. Else, find it by tag
+        if not uid:
+            tag = tag or "latest"
+            tag = self._query(
+                session, FeatureSet.Tag, project=project, name=tag, obj_name=name
+            ).one_or_none()
+            if not tag:
+                raise mlrun.errors.MLRunNotFoundError(
+                    f"Feature-set not found {project}/{name}:{tag}"
+                )
+            query = self._query(session, FeatureSet, id=tag.obj_id)
+        else:
+            query = self._query(
+                session, FeatureSet, name=name, project=project, uid=uid
+            )
+
+        db_fs = query.one_or_none()
+        if not db_fs:
+            raise mlrun.errors.MLRunNotFoundError(
+                f"Feature-set not found {project}/{name}:{tag}"
+            )
+
+        data = fs.copy()
+        stat = data.pop("status", None)
+        if stat:
+            state = stat.get("status")
+            db_fs.state = state or db_fs.state
+            # TODO - granular update of JSON fields.
+            db_fs.status = stat
```

Comment:
```suggestion
        db_fs.status = data.get("status", db_fs.status)
        db_fs.state = data.get("status", {}).get("state", db_fs.state)        
```

Reply:
Suggested code doesn't work. `data` has a key called `status` whose value is `None`. This means it does not substitute the default value for it on `get()`, but rather returns None. Calling another `get()` on that in line 2 throws an exception. Line 1 doesn't work as well, since the default value doesn't get substituted.